go/test/endtoend/tabletmanager/lock_unlock_test.go

```diff
@@ -0,0 +1,207 @@
+/*
+Copyright 2019 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tabletmanager
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"vitess.io/vitess/go/mysql"
+)
+
+// TestLockAndUnlock tests the lock ability by locking a replica and asserting it does not see changes
+func TestLockAndUnlock(t *testing.T) {
+	ctx := context.Background()
+
+	masterConn, err := mysql.Connect(ctx, &masterTabletParams)
+	if err != nil {
```

Comment:
We should use testify for this kind of stuff. Instead of this `if` business, just write `require.NoError(t, err)`